src/contexts/Api/index.tsx

```diff
@@ -66,36 +55,14 @@ export const APIProvider = ({ children }: { children: React.ReactNode }) => {
     !!localStorage.getItem('isLightClient')
   );
 
-  // initial connection
   useEffect(() => {
-    if (!provider) {
-      const _network: NetworkName = localStorage.getItem(
-        'network'
-      ) as NetworkName;
-      connect(_network, isLightClient);
-    }
-  });
-
-  // provider event handlers
-  useEffect(() => {
-    if (provider !== null) {
-      provider.on('connected', () => {
-        setConnectionStatus(ConnectionStatus.Connected);
-      });
-      provider.on('error', () => {
-        setConnectionStatus(ConnectionStatus.Disconnected);
-      });
-      connectedCallback(provider);
-    }
-  }, [provider]);
+    connect(networkName, isLightClient);
+  }, []);
```

Comment:
It took me some time to figure out why there are no `deps` here.
Perhaps a clearer solution would be to do:
```
  useEffect(() => {
    connect(NETWORKS[initialNetworkName], localStorage.getItem('isLightClient'));
  }, [initialNetworkName]);
```
or to add some comments?

Reply:
The thing with `isLightClient` is that it's used in 1+ places, so `localStorage.getItem('isLightClient')` would have to be repeated. Similar with `networkName`. Reading deeper the code, I reckon that the cleanest solution would be to have a `useEffect` with `networkName` and `isLightClient` as deps, reconnecting every time they change, but there are different paths those values are modified through so I don't feel comfortable rewiring everything atm without a thorough testing, which we don't have time for right now.